crates/core/src/fnc/array.rs

```diff
@@ -603,7 +616,8 @@ pub fn remove((mut array, mut index): (Array, i64)) -> Result<Value, Error> {
 	Ok(array.into())
 }
 
-pub fn repeat((value, count): (Value, usize)) -> Result<Value, Error> {
+pub fn repeat((value, count): (Value, i64)) -> Result<Value, Error> {
+	let count = count.max(0) as usize;
```

Comment:
Rather than implicitly maxxing this to `0` when an input is negative, wouldn't it be less surprising to the end user to instead return an error if they give a negative value for `count`? I was looking at the `.surql` test and having `repeat(1, -256)` result in `[]` seemed very surprising to me.

Reply:
I agree that that is probably the better option. Previously it was just casting the i64 to usize so -256 would result in a very large number/array and it would error with too large. This was a quick change because the PR was already getting very large. I'll revert it to the previous behavior and add a note that we should look at the casting done there like in the other places I found improper integer conversions.